backend/src/web_handler.py

```diff
@@ -0,0 +1,120 @@
+from functools import wraps
+
+import jwt
+from flask import Flask, session, Response, request
+from flask_cors import CORS
+from flask_restful import Api, Resource
+from pony.orm import db_session
+from pony.orm.serialization import to_dict
+
+from config import gamma_config as config
+from db import Tag
+from process.GammaProcess import handle_gamma_me, handle_gamma_auth, handle_gamma_signout
+from process.SongProcess import handle_get_songs_and_tags, handle_get_song_by_id, handle_delete_song, \
+    handle_create_song, handle_update_song
+from process.TagProcess import handle_get_tags, handle_delete_tag, handle_get_tag_by_id
+
+app = Flask(__name__)
+api = Api(app)
+
+cors = CORS(app, resources={r"/*": {"origins": "*"}})
+app.config['CORS_HEADERS'] = 'Content-Type'
+
+app.secret_key = config.SECRET_KEY
+
+
+def admin_required(f):
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        if "token" in session:
+            authorities = jwt.decode(jwt=session['token'], options={'verify_signature': False})["authorities"]
+
+            if config.GAMMA_ADMIN_AUTHORITY not in authorities:
+                return Response(status=403)
+        return f(*args, **kwargs)
+
+    return decorated_function
+
+
+class SongRes(Resource):
+    @db_session
```

Comment:
Preferably don't use @db_sessions outside of command/query files where possible, only having these files communicating with the db (and doing nothing/little else) gives a clean and easy-to-find interface where all database communication is done.